sky/packages/sky/lib/src/widgets/drawer.dart

```diff
@@ -132,41 +130,21 @@ class Drawer extends StatefulComponent {
 
   void _settle() { _isMostlyClosed ? _performance.reverse() : _performance.play(); }
 
-  // TODO(mpcomplete): Figure out how to generalize these handlers on a
-  // "PannableThingy" interface.
-  EventDisposition handlePointerDown(_) {
-    _performance.stop();
-    return EventDisposition.processed;
- }
-
-  EventDisposition handlePointerMove(sky.PointerEvent event) {
+  void _handleDragUpdate(double delta) {
     if (_performance.isAnimating)
-      return EventDisposition.ignored;
-    _performance.progress += event.dx / _kWidth;
-    return EventDisposition.processed;
-  }
-
-  EventDisposition handlePointerUp(_) {
-    if (!_performance.isAnimating) {
-      _settle();
-      return EventDisposition.processed;
-    }
-    return EventDisposition.ignored;
+      return;
+    _performance.progress += delta / _kWidth;
   }
 
-  EventDisposition handlePointerCancel(_) {
-    if (!_performance.isAnimating) {
+  void _handleDragEnd(Offset velocity) {
+    if (_performance.isAnimating)
+      return;
+    if (velocity == Offset.zero) {
       _settle();
-      return EventDisposition.processed;
+    } else {
+      if (velocity.dx.abs() >= _kMinFlingVelocity)
+        _performance.fling(velocity: velocity.dx * _kFlingVelocityScale);
```

Comment:
this doesn't seem right. Surely when you release a horizontal drag on a drawer, it should always either open or close.